components/table/src/simple-columns/favorite-column.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Optional,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { DtTable } from '../table';
+import { DtSimpleColumnBase } from './simple-column-base';
+
+@Component({
+  selector: 'dt-favorite-column',
+  templateUrl: 'favorite-column.html',
+  styleUrls: ['./favorite-column.scss'],
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.Emulated,
+  /*
+   * Deliberatley set to Default because the embedded view that gets created for the
+   * dtColumDef can't handle onPush and results in ChangeAfterChecked error.
+   */
+  changeDetection: ChangeDetectionStrategy.Default,
+  providers: [{ provide: DtSimpleColumnBase, useExisting: DtFavoriteColumn }],
+})
+export class DtFavoriteColumn<T> extends DtSimpleColumnBase<T> {
+  @Output() readonly change = new EventEmitter<T>();
+
+  constructor(@Optional() table: DtTable<T>) {
+    super(table);
+  }
+
+  /**
+   * @internal Get data either returns a data access with the given name or calls the
+   * displayAccessor function to get the simpleData for display.
+   */
+  // tslint:disable-next-line: no-any
+  _getData(data: T): any {
```

Comment:
Could use a property instead of this function and update it only when needed. because right now, this function is called 3 times in the template on every change detection run. This means it gets called a lot even if the result does not change.

Reply:
Hi Thomas, I improved the template a bit but I tried to apply completely your suggestion but it seems it does not cover fully our case.
The main issue is that we need to update the reference in the template when clicking on the icon but if we do without _`getData` I always get the old reference (even by triggering the change detection manually does not solve this issue).
For example if the first row has favorite=true and the second=false, I always get the first row as it's false and the second as it's true.
How can I get the proper value without accessing the row?
Do you have any suggestion to solve this issue?
I tried to look also for other usages but in all the cases we always accessing through _getData (also more than one time as in our case)

Reply:
@yngrdyn thanks for clarification. This makes now totally sense. LGTM :)

Only thing is, can me rename `getData` because as far as i tries to always return a `string` (title) for displaying it in the template. Maybe we could then also change the return type too?

Reply:
I renamed it to `_isFavorite` and now returns a boolean